src/components/list/SearchableList.tsx

```diff
@@ -0,0 +1,213 @@
+import React, { FC, useCallback, useEffect, useState } from 'react';
+
+import Fuse, { FuseResult } from 'fuse.js';
+
+import { SearchField, SearchFieldOptions } from '../input/SearchField';
+import Flex from '../flex';
+import { ListItemType, SearchableListItem } from './SearchableListItem';
+
+import List from './';
+
+export type ListItemWithChildren = ListItemType & {
+  children?: Array<ListItemType>;
+};
+
+export type Props = {
+  listItems: Array<ListItemWithChildren>;
+  noResults?: FC;
+  insertion?: FC;
+  ariaLabel?: string;
+  searchFieldOptions?: SearchFieldOptions;
+  listOptions?: { columns?: number; childrenSpacing?: 'md' | '2xl' };
+};
+
+type FuseSearch = Fuse<ListItemWithChildren> & {
+  options: typeof fuseOptions;
+};
+
+const fuseOptions = {
+  keys: ['label', 'children.label'],
+  includeMatches: true,
+  threshold: 0,
+  ignoreLocation: true,
+  minMatchCharLength: 1,
+  shouldSort: false,
+};
+
+const mapItemsFromSearchResult = (
+  searchResults: FuseResult<ListItemWithChildren>[],
+): Array<ListItemWithChildren> => {
```

Comment:
Is there a difference?

Reply:
No, it's for consistency. Personally, I tend to use `Array` only when I already have a complex type inside (like a record of records) when it would be hard to figure out to which type the `[]` refers.